static/js/stream_list.js

```diff
@@ -108,13 +108,18 @@ exports.build_stream_list = function () {
 
     _.each(stream_groups.pinned_streams, add_sidebar_li);
 
-    if (stream_groups.pinned_streams.length > 0) {
+    // true when at least one stream is in this category
+    var any_pinned_streams = stream_groups.pinned_streams.length;
+    var any_normal_streams = stream_groups.normal_streams.length;
+    var any_dormant_streams = stream_groups.dormant_streams.length ;
```

Comment:
Do actual comparisons here (e.g. `.length > 0`).  While the old code should work, it'd be a lot more readable that these are bools with the comparison.

Reply:
That was how I did it originally before a was told to 'simplify' the expression. But I agree with you; I will revert the changes.